src/svd/restriction.rs

```diff
@@ -1,79 +1,80 @@
+    Vec<svd_rs::EnumeratedValues>,
+)> {
+    let restrictions = match restriction {
+        // No write constraint, the constraint is specified by the size of the
+        // field in bits.
+        //
+        // A software that generates register access from this should generate
+        // a correct bit mask.
+        chip::ValueRestriction::Any => (
+            if width > 1 {
+                Some(svd_rs::WriteConstraint::Range(
+                    svd_rs::WriteConstraintRange {
+                        min: 0,
+                        max: 2u64.pow(width) - 1,
+                    },
+                ))
             } else {
-                let mut el = xmltree::Element::new("writeConstraint");
-                let mut range = xmltree::Element::new("range");
-                range.child_with_text("minimum", 0.to_string());
-                range.child_with_text("maximum", (2usize.pow(width as u32) - 1).to_string());
-                el.children.push(range);
-                vec![el]
-            }
-        }
-        chip::ValueRestriction::Range(lo, hi) => {
-            let mut el = xmltree::Element::new("writeConstraint");
-            let mut range = xmltree::Element::new("range");
-            range.child_with_text("minimum", lo.to_string());
-            range.child_with_text("maximum", hi.to_string());
-            el.children.push(range);
-            vec![el]
-        }
-        chip::ValueRestriction::Enumerated(enumerated) => {
-            let mut wc = xmltree::Element::new("writeConstraint");
-            wc.child_with_text("useEnumeratedValues", "true");
-
-            let mut values_el = xmltree::Element::new("enumeratedValues");
+                None
+            },
+            vec![],
+        ),
+        chip::ValueRestriction::Range(ref lo, ref hi) => (
+            Some(svd_rs::WriteConstraint::Range(
+                svd_rs::WriteConstraintRange { min: *lo, max: *hi },
+            )),
+            vec![],
+        ),
+        chip::ValueRestriction::Enumerated(ref v) => {
+            let mut values = v.values().collect::<Vec<_>>();
+            values.sort_by(|a, b| a.value.cmp(&b.value));
 
-            let mut values: Vec<_> = enumerated.values().collect();
-            values.sort_by(|v1, v2| v1.value.cmp(&v2.value));
-            values_el.children = values
+            let values = values
                 .into_iter()
-                .inspect(|v| {
-                    if v.name.starts_with("VAL_0x") {
-                        log::info!("Undescriptive name {:?} in {:?}", v.name, name);
-                    }
-                })
                 .map(generate_enumerated)
-                .collect::<Result<_, _>>()?;
+                .collect::<Result<Vec<_>, _>>()?;
 
-            vec![wc, values_el]
+            let enumerated_values = svd_rs::EnumeratedValues::builder()
+                .values(values)
+                .build(svd_rs::ValidateLevel::Strict)
+                .unwrap();
```

Comment:
I agree with the approach, the unwrap was probably a left over since I did all the code using unwrap first then added error handling. I'll change that.

Reply:
Solved, also removed another unwrap that I left on the encode function.